### PR TITLE
Feature forex rates

### PR DESCRIPTION
**Pull request by EugZol**
Added fiat exchange rate providers (one provider for now).

The logic of exchange is the following (answering the question 'how much BTC is in X (RUB, CAD, etc.)):
1. Check BTC exchanges for X->BTC rates. If found, return the result.
2. Check BTC exchanges for USD->BTC rates. If not found, return error.
3. Check fiat exchanges for X->USD rates. If not found, return error.
4. Return cross-rate X->USD->BTC.

### CODE REVIEW

File: lib/straight/gateway.rb [hunk 1]
```diff
@@ -156,8 +157,18 @@ def amount_from_exchange_rate(amount, currency:, btc_denomination: :satoshi)
           return Satoshi.new(amount, from_unit: btc_denomination).to_i
         end
 
-        try_adapters(@exchange_rate_adapters, type: "exchange rate") do |a|
-          a.convert_from_currency(amount, currency: currency)
+        begin
+          try_adapters(@exchange_rate_adapters, type: "exchange rate", priority_exception: Straight::ExchangeRate::Adapter::CurrencyNotSupported) do |a|
```
Comment (dsnipe) [inline lib/straight/gateway.rb]: A little long string :)

Comment (EugZol) [inline lib/straight/gateway.rb]: Sorry, ultra-wide screen makes me miss such things :) Will change.

File: lib/core_ext/deep_get.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  def deep_get(*args)
+    args.inject(self) do |intermediate, key|
+      if intermediate.respond_to?(:[])
+        intermediate[key]
+      else
+        return nil
+      end
+    end
+  end
+end
```
Comment (AlexeyShepelev) [inline lib/core_ext/deep_get.rb]: No newline at end of file

File: lib/straight/exchange_rate_adapter.rb [hunk 3]
```diff
@@ -57,8 +39,6 @@ def get_rate_value_from_hash(rates_hash, *keys)
       def rate_to_f(rate)
         rate ? rate.to_f : raise(CurrencyNotSupported)
       end
-
     end
-
   end
-end
+end
```
Comment (AlexeyShepelev) [inline lib/straight/exchange_rate_adapter.rb]: No newline at end of file

**Pull request merged**